### DISCUSSION TITLE
redirect http to https

### DISCUSSION THREAD

**Discussion by kassemchakaroun** [gravitational/teleport#35965] [Q&A] [answered]
hello
hope you are all doing well !!
i install teleport in my lab and i want to redirect connection from http to https
like from http://teleport.labo.local to https://teleport.labo.local

this is my teleport.yaml configuration 

version: v3
teleport:
  nodename: teleport
  data_dir: /var/lib/teleport
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  cluster_name: teleport.labo.local
  proxy_listener_mode: multiplex
ssh_service:
  enabled: "yes"
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  enabled: "yes"
  web_listen_addr: 0.0.0.0:443
  public_addr: teleport.labo.local:443
  https_keypairs:
  - key_file: /var/lib/teleport/teleport.key
    cert_file: /var/lib/teleport/teleport.cer
  https_keypairs_reload_interval: 0s
  acme: {}

installed on rocky linux 8

**Comment by webvictim** [comment, marked as answer]
Teleport doesn't support that itself, see these old issues:
- https://github.com/gravitational/teleport/issues/3853
- https://github.com/gravitational/teleport/issues/7500

The easiest way to achieve this is to put another web server like nginx or Caddy in front of Teleport and have that send redirects to HTTPS when it receives HTTP requests.

There is a guide here on how to set up Teleport behind nginx, which will automatically configure this behaviour: https://github.com/gravitational/teleport/discussions/26445
